Makefile.kokkos

```diff
@@ -667,7 +667,7 @@ ifeq ($(KOKKOS_INTERNAL_USE_CUDA), 1)
 
   ifeq ($(KOKKOS_INTERNAL_COMPILER_NVCC), 1)
     tmp := $(call kokkos_append_header,"$H""define KOKKOS_ENABLE_CUDA_LAMBDA")
-    KOKKOS_CXXFLAGS += -expt-extended-lambda
+    KOKKOS_CXXFLAGS += -extended-lambda
```

Comment:
The `-extended-lambda` flag is supported for all CUDA versions that we support with NVCC.

Reply:
can we write what you said here as a comment there so that it stays as reminder for later?

Reply:
Unnecessary, I think, since the alternative is entirely backwards-looking in time. It's not something that's going to change or need re-consideration.

This is what writing things down in the review is for